x-pack/plugins/cloud/public/plugin.ts

```diff
@@ -198,7 +211,32 @@ export class CloudPlugin implements Plugin<CloudSetup> {
       if (userId) {
         // Do the hashing here to keep it at clear as possible in our source code that we do not send literal user IDs
         const hashedId = sha256(userId.toString());
-        fullStory.identify(hashedId);
+        application?.then(async () => {
+          try {
+            const appStart = await application;
+            this.appSubscription = appStart.currentAppId$.subscribe((appId) => {
+              // Update the current application every time it changes
+              fullStory.setUserVars({
+                app_id_str: appId ?? 'unknown',
+              });
+            });
+          } catch (e) {
+            // eslint-disable-next-line no-console
+            console.error(
+              `[cloud.full_story] Could not retrieve application service due to error: ${e.toString()}`,
+              e
+            );
+          }
```

Comment:
NIT: would move the catch statement to the `then` handler (`application?.then(..., () => { ... }`)